get_s3.go

```diff
@@ -204,29 +212,45 @@ func (g *S3Getter) getAWSConfig(region string, creds *credentials.Credentials) *
 }
 
 func (g *S3Getter) parseUrl(u *url.URL) (region, bucket, path, version string, creds *credentials.Credentials, err error) {
```

Comment:
Lets add  some unit tests to test parsing both s3 and s3 complaint URLs. Lets us get some test coverage and avoid running the server in tests.

After that we can merge!